Documentation/api-guidelines/System.Memory.md

```diff
@@ -0,0 +1,46 @@
+# System.Memory Design Guidelines
+
+`System.Memory` is a collection of types and features that make working with
+buffers and raw memory more efficient while remaining type safe. The feature
+specs can be found here:
+
+* [`Span<T>`](https://github.com/dotnet/corefxlab/blob/master/docs/specs/span.md)
```

Comment:
Should we move these spec docs to the corefx repo?

Reply:
Maybe, but I don't think we should have an expectation that these are the official source of truth at this point.

Reply:
I think there is a difference between spec and docs. The spec talks about details that are not interesting to many people who just want to use the type.

Reply:
Agreed. I think during initial design of the API Guidance having the specs linked helps as specs also convey the design goals and desired constraints.